### PR TITLE
Docs: Add logs data source guide

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

Adds a guide on how to enable logs support for a data source plugin.

### CODE REVIEW

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+   Just like for metrics data, Grafana expects your plugin to return log data as a [data frame]({{< relref "data-frames.md" >}}). Grafana considers any data frame with at least one time field and one text field to be log data.
+
+   ```ts
+   return new MutableDataFrame({
+     refId: query.refId,
+     fields: [
+       {
+         name: 'time',
+         type: FieldType.time,
+         values: [from, to]
```
Comment (aocenas) [inline docs/sources/developers/plugins/data-sources-for-logs.md:34]: The from/to is kinda confusing, looks like it should be a range, but it's just timestamp1, timestamp2 basically

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    },
+  ],
+});
+```
+
+## Parsed fields
+
+You can add additional information about each log line by adding more data frame fields.
+
+If a data frame has more than one text field, Grafana assumes the first field in the data frame to be the actual log line. Any subsequent text fields are treated as [parsed fields]({{< relref "../../features/explore.md#labels-and-parsed-fields" >}}).
```
Comment (aocenas) [inline docs/sources/developers/plugins/data-sources-for-logs.md:81]: tbh I don't even know right now :D. We do that for elastic but not sure if also for other DS but if you checked then it makes sense.

File: docs/sources/developers/plugins/data-sources-for-logs.md (hunk 1 above)
Comment (marcusolsson) [reply docs/sources/developers/plugins/data-sources-for-logs.md:34]: This is from the grafana/toolkit scaffold. I'll change to hard-coded timestamp 👍

File: docs/sources/developers/plugins/data-sources-for-logs.md (hunk 2 above)
Comment (marcusolsson) [reply docs/sources/developers/plugins/data-sources-for-logs.md:81]: That's the behavior I'm seeing at least :P

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+     "logs": true
+   }
+   ```
+
+1. **Construct the log data**.
+
+   Just like for metrics data, Grafana expects your plugin to return log data as a [data frame]({{< relref "data-frames.md" >}}). Grafana considers any data frame with at least one time field and one text field to be log data.
+
+   ```ts
+   return new MutableDataFrame({
```
Comment (ryantxu) [inline docs/sources/developers/plugins/data-sources-for-logs.md:28]: MutableDataFrame is only useful if you actually mutate :)  so maybe an example like this is better?

```

    const frame = new MutableDataFrame({
      refId: 'a',
      fields: [
        { name: 'time', type: FieldType.time },
        { name: 'line', type: FieldType.string },
      ],
    });
    frame.add({ time: 12345, line: 'user registered' });
    frame.add({ time: 12345, line: 'user logged in' });
```

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
++++
+title = "Build a logs data source"
+type = "docs"
++++
+
+# Build a logs data source
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:6]: Please change the file name to match the title.

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+This guide explains how to build a logs data source plugin.
+
+Data sources in Grafana supports both metrics and log data. The steps to build a logs data source are largely the same as for a metrics data source. This guide assumes that you're already familiar with how to [Build a data source plugin]({{< relref "../../../../../tutorials/build-a-data-source-plugin.md" >}}) for metrics.
+
+
+1. **Enable logs support**.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:13]: Periods should be inside the bolding. In most cases, associated punctuation should be formatted the same as the text.

That said, I think making these these steps AND bold is overkill. What about an H2 to hold both of them and call out the task, then make them both H3? Either that or change both steps to h2.

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+   Let Grafana know that your data source plugin can return log data, by adding `"logs": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:15]:
```suggestion
   Tell Grafana that your data source plugin can return log data by adding `"logs": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
```

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+   ```json
+   {
+     "logs": true
+   }
+   ```
+
+1. **Construct the log data**.
+
+   Just like for metrics data, Grafana expects your plugin to return log data as a [data frame]({{< relref "data-frames.md" >}}). Grafana considers any data frame with at least one time field and one text field to be log data.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:25]: This is not telling me what to do. I kind of infer that I need to put the log data in a data frame? If so, you should state that.

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+       {
+         name: 'content',
+         type: FieldType.string,
+         values: ['user registered', 'user logged in']
+       },
+     ],
+   });
+   ```
+
+That's all you need to start returning log data from your data source. Go ahead and try it out in [Explore]({{< relref "../../features/explore.md#explore" >}}), or by adding a [Logs panel]({{< relref "../../features/panels/logs.md#logs-panel" >}}).
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:45]:
```suggestion
That's all you need to start returning log data from your data source. Go ahead and try it out in [Explore]({{< relref "../../features/explore.md#explore" >}}) or by adding a [Logs panel]({{< relref "../../features/panels/logs.md#logs-panel" >}}).
```

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+Congratulations, you just wrote your first logs data source plugin! Next, let's look at a couple of features that can improve the experience for the user.
+
+
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:49]:
```suggestion
```

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+## Labels
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:50]: I suggest changing this heading and the next to "Verb nouns" or "Verb a noun" to maintain consistency with the earlier headings/steps.

File: docs/sources/developers/plugins/data-sources-for-logs.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+To help filter log lines, many log systems let you query logs based on metadata, or _labels_.
+
+You can add labels to a stream of logs by setting the [labels](../../packages_api/data/field.md#labels-property) property on the [Field]({{< relref "../../packages_api/data/field.md" >}}).
+
+**Example**:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:56]: See earlier comments about punctuation.

File: docs/sources/developers/plugins/data-sources-for-logs.md (hunk 2 above)
Comment (oddlittlebird) [inline docs/sources/developers/plugins/data-sources-for-logs.md:81]:
```suggestion
If a data frame has more than one text field, then Grafana assumes the first field in the data frame to be the actual log line. Any subsequent text fields are treated as [parsed fields]({{< relref "../../features/explore.md#labels-and-parsed-fields" >}}).
```

Comment (oddlittlebird) [reply docs/sources/developers/plugins/data-sources-for-logs.md:81]: If you use an "if," also include a "then." This helps people with lower-level English skills.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
This is great content, and desperately needed!

**Review by oddlittlebird — APPROVED** [review]
Looks great!

**Pull request merged**